tests/test_multistore_file.py

```diff
@@ -41,3 +74,170 @@ def test_lock_file_raises_ioerror(self):
+            try:
+                store.get()
+                self.fail('Should have raised an exception.')
+            except locked_file.CredentialsFileSymbolicLinkError:
+                pass
+            finally:
+                os.unlink(SYMFILENAME)
+
+    def test_multistore_non_existent_file(self):
+        store = multistore_file.get_credential_storage(
+            FILENAME,
+            'some_client_id',
+            'user-agent/1.0',
+            ['some-scope', 'some-other-scope'])
+
+        credentials = store.get()
+        self.assertEquals(None, credentials)
+
+    def test_multistore_file(self):
+        credentials = self.create_test_credentials()
+
+        store = multistore_file.get_credential_storage(
+            FILENAME,
+            credentials.client_id,
+            credentials.user_agent,
+            ['some-scope', 'some-other-scope'])
+
+        store.put(credentials)
+        credentials = store.get()
+
+        self.assertNotEquals(None, credentials)
+        self.assertEquals('foo', credentials.access_token)
+
+        store.delete()
+        credentials = store.get()
+
+        self.assertEquals(None, credentials)
+
+        if os.name == 'posix':
+            self.assertEquals(
+                0o600, stat.S_IMODE(os.stat(FILENAME).st_mode))
+
+    def test_multistore_file_custom_key(self):
+        credentials = self.create_test_credentials()
+
+        custom_key = {'myapp': 'testing', 'clientid': 'some client'}
+        store = multistore_file.get_credential_storage_custom_key(
+            FILENAME, custom_key)
+
+        store.put(credentials)
+        stored_credentials = store.get()
+
+        self.assertNotEquals(None, stored_credentials)
+        self.assertEqual(credentials.access_token,
+                         stored_credentials.access_token)
+
+        store.delete()
+        stored_credentials = store.get()
+
+        self.assertEquals(None, stored_credentials)
+
+    def test_multistore_file_custom_string_key(self):
+        credentials = self.create_test_credentials()
+
+        # store with string key
+        store = multistore_file.get_credential_storage_custom_string_key(
+            FILENAME, 'mykey')
+
+        store.put(credentials)
+        stored_credentials = store.get()
+
+        self.assertNotEquals(None, stored_credentials)
+        self.assertEqual(credentials.access_token,
+                         stored_credentials.access_token)
+
+        # try retrieving with a dictionary
+        multistore_file.get_credential_storage_custom_string_key(
+            FILENAME, {'key': 'mykey'})
+        stored_credentials = store.get()
+        self.assertNotEquals(None, stored_credentials)
+        self.assertEqual(credentials.access_token,
+                         stored_credentials.access_token)
+
+        store.delete()
+        stored_credentials = store.get()
+
+        self.assertEquals(None, stored_credentials)
+
+    def test_multistore_file_backwards_compatibility(self):
```

Comment:
What purpose do we have to provide backwards compat?

Doing a `git blame` for the `backwards` compat goes back to 34c1ff543dd16edf58dcf5b336076cf27de3721a

Before then it was last touched in e2233cd7a27247dbad0d06cc25972bd4bec03d14 where it was introduced when the multistore implementation changed.